pkg/api/util/ipc_endpoint_test.go

```diff
@@ -0,0 +1,212 @@
+	assert.NoError(t, err)
+	assert.True(t, end.closeConn)
+}
+
+func TestIPCEndpointGet(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	// test that DoGet will hit the endpoint url
+	res, err := end.DoGet()
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api")
+}
+
+func TestIPCEndpointGetWithValues(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	// set url values for GET request
+	v := url.Values{}
+	v.Set("verbose", "true")
+
+	// test construction with option for url.Values
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	// test that DoGet will use query parameters from the url.Values
+	res, err := end.DoGet(WithValues(v))
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api?verbose=true")
+}
+
+func TestIPCEndpointGetWithHostAndPort(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	// modify the config so that it uses a different setting for the cmd_host
+	conf.Set("process_config.cmd_host", "localhost", pkgconfigmodel.SourceAgentRuntime)
+
+	// test construction with alternate values for the host and port
+	end, err := NewIPCEndpoint(conf, "test/api", WithHostAndPort(conf.GetString("process_config.cmd_host"), conf.GetInt("cmd_port")))
+	assert.NoError(t, err)
+
+	res, err := end.DoGet()
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api")
+}
+
+func TestIPCEndpointDeprecatedIPCAddress(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	// Use the deprecated (but still supported) option "ipc_address"
+	conf.UnsetForSource("cmd_host", pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("ipc_address", "localhost", pkgconfigmodel.SourceAgentRuntime)
+
+	// test construction, uses ipc_address instead of cmd_host
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	res, err := end.DoGet()
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api")
```

Comment:
Should we test the full URL here as well?

Reply:
Yes, good idea.